addons/website/models/website.py

```diff
@@ -1301,18 +1301,24 @@ def get_client_action_url(self, url, mode_edit=False):
             action_params["enable_editor"] = 1
         return "/web#" + urls.url_encode(action_params)
 
+    def get_client_action(self, url, mode_edit=False, website_id=False):
+        action = self.env["ir.actions.actions"]._for_xml_id("website.website_preview")
```

Comment:
Nice :+1: That actually was my only fear making this method, not using the XML-defined action.